proposals/exception-handling/Exceptions-formal-examples.md

```diff
@@ -0,0 +1,192 @@
+
+Its purpose is to make sure everyone is happy with the implications of the semantics in the current 3rd proposal, or to aid discussions on these semantics.
+
+The first *example 0* contains all the new instructions, and it is the only one with an almost full reduction displayed. It is meant to easily show how the spec works, even if the reader has not spent much time with the WebAssembly formal spec.
+
+For all other examples just the result of the reduction is given. These examples are taken from comments in this repository, which are linked. Some times/often the examples are modified to fit the current syntax.
+
+If anyone would like that I add another reduction trace, or other examples, please let me know, I'd be happy to.
+
+### notation
+
+If `x` is an exception tag index, then `a_x` denotes its exception tag address, i.e., `F.exception[x] = a_x`, where `F` is the current frame.
+
+## example 0
+
+The only example with an almost full reduction trace, and all new instructions. The first 3 steps, reducing the several `try`s to their respective administrative instructions, are not shown.
+
+```
+(func (result i32) (local i32)
+  try
+    try
+      try
+        throw x
+      catch_all
+        i32.const 27
+        local.set 0
+      rethrow 0
+      end
+    delegate 0
+  catch x
+    local.get 0
+  end)
+```
+
+Take the frame `F = (locals i32.const 0, module m)`. We have:
+
+```
+↪ ↪ ↪ F; label_1{} (catch_1{a_x local.get 0}
+           (label_0{} (delegate{0}
+             (label_0{} (catch_0{i32.const 27 local.set 0 rethrow 0}
+               throw a_x end) end) end) end) end) end
+```
+
+For the empty throw context `T = [_]` the above is the same as
+
+```
+F; label_1{} (catch_1{a_x local.get 0}
+     label_0{} (delegate{0}
+       label_0{} (catch_0{i32.const 27 local.set 0 rethrow 0}
+         T[throw a_x] end) end) end) end) end
+
+↪ F; label_1{} (catch_1{a_x local.get 0}
+       (label_0{} (delegate{0}
+         (label_0{} (caught{a_x} i32.const 27 local.set 0 rethrow 0
+           end) end) end) end) end) end
+```
+
+Let `F'` be the frame `{locals i32.const 27, module m}`, and let `B^0 = [_]`.
+
+```
+↪ F; label_1{} (catch_1{a_x local.get 0}
+       (label_0{} (delegate{0}
+         (label_0{} (caught{a_x} B^0 [rethrow 0]
+           end) end) end) end) end) end
+
+↪ F; label_1{} (catch_1{a_x local.get 0}
+       (label_0{} (delegate{0}
+         (label_0{} (caught{a_x} B^0 [throw a_x]
+           end) end) end) end) end) end
+```
+
+Let `T' = label_0{} (caught{a_x} B^0 [_] end) end`.
+
+```
+↪ F; label_1{} (catch_1{a_x local.get 0}
+       (label_0{} (delegate{0}
+         T'[throw a_x] end) end) end) end
+
+↪ F; label_1{} (catch_1{a_x local.get 0}
+       (label_0{} (delegate{0}
+         T'[throw a_x] end) end) end) end
+
+↪ F; label_1{} (catch_1{a_x local.get 0}
+       (label_0{} (delegate{0}
+         T'[throw a_x] end) end) end) end
+
+↪ F; label_1{} (catch_1{a_x local.get 0}
+       T'[throw a_x] end) end
```

Comment:
Hm, there seems to be a step missing where the delegate is reduced via some B^0 but the inner label is still present.